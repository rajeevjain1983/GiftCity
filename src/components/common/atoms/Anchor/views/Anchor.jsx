// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import {
  buildUrl,
  getAsPathWithSlug,
  getMappedPageHref,
  triggerPostMessage,
  getAPIConfig,
} from '../../../../../utils';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

const getAsLinkPath = (IsSlugPathAdded, asPath, incomingUrl) => {
  return IsSlugPathAdded ? asPath || incomingUrl : getAsPathWithSlug(asPath || incomingUrl);
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  asPath,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  url,
  text,
  dataLocator,
  IsSlugPathAdded,
  ...other
}) => {
  const targetVal = target || '_self';
  let incomingUrl = to || url;
  const asLinkPath = getAsLinkPath(IsSlugPathAdded, asPath, incomingUrl);
  if (!noLink) {
    incomingUrl = getMappedPageHref(incomingUrl);
  }
  const hrefUrl = asLinkPath || buildUrl(incomingUrl);

  const apiConfigObj = getAPIConfig();

  let AnchorComponent = null;
  if (children || text) {
    AnchorComponent = noLink ? (
      <a
        href={buildUrl(incomingUrl)}
        className={className}
        onClick={handleLinkClick}
        title={title}
        target={targetVal}
        data-locator={dataLocator}
        {...other}
      >
        {children}
      </a>
    ) : (
      <Link href={incomingUrl} as={asLinkPath} shallow={shallow} scroll={scroll}>
        <a
          className={className}
          href={hrefUrl}
          title={title}
          target={targetVal}
          {...(apiConfigObj.isAppChannel
            ? {
                onClick: () => triggerPostMessage(to || asLinkPath),
              }
            : '')}
          data-locator={dataLocator}
          {...other}
        >
          {children || text}
        </a>
      </Link>
    );
  }
  return AnchorComponent;
};

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  asPath: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.string,
  dataLocator: PropTypes.string,
  IsSlugPathAdded: PropTypes.bool,
};

Anchor.defaultProps = {
  asPath: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
  url: '',
  text: '',
  dataLocator: '',
  IsSlugPathAdded: false,
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
