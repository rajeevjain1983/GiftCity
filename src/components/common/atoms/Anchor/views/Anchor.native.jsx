// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity } from 'react-native';
import {
  setTestId,
  getLocator,
  configureInternalNavigationFromCMSUrl,
  isWebViewPage,
} from '@tcp/core/src/utils';
import { StyledText } from '../../../../../../styles/globalStyles/StyledText.style';
import {
  UrlHandler,
  navigateToPage,
  validateExternalUrl,
  redirectToInAppView,
} from '../../../../../utils/index.native';
import withStyles from '../../../hoc/withStyles.native';
import { AnchorStyles, AnchorView, AnchorIcon } from '../Anchor.style.native';
import AccessibilityRoles from '../../../../../constants/AccessibilityRoles.constant';

const Icon = require('../../../../../../../mobileapp/src/assets/images/carrot-small-rights.png');

const getOverrideStyle = overrideStyle => {
  return overrideStyle && overrideStyle.color ? { style: overrideStyle } : null;
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is a Anchor component to manage the internal or external .
 */
const Anchor = ({
  anchorVariation,
  text,
  visible,
  children,
  customStyle,
  locator,
  onPress,
  accessibilityLabel,
  justifyContent,
  hitSlop,
  overrideStyle = {},
  ...otherProps
}) => {
  const { url, navigation, onReuqestClose } = otherProps;
  const accessibilityRole = validateExternalUrl(url)
    ? AccessibilityRoles.Link
    : AccessibilityRoles.Button;
  const openUrl = () => {
    if (isWebViewPage(url) && navigation) {
      if (onReuqestClose) {
        onReuqestClose();
      }
      redirectToInAppView(url, navigation);
    } else if (validateExternalUrl(url)) {
      UrlHandler(url);
    } else if (navigation) {
      const cmsValidatedUrl = configureInternalNavigationFromCMSUrl(url);
      navigateToPage(cmsValidatedUrl, navigation);
    }
  };
  const styleProp = getOverrideStyle(overrideStyle);

  if (children) {
    return (
      <TouchableOpacity
        accessibilityRole={accessibilityRole}
        onPress={onPress || openUrl}
        {...otherProps}
        style={customStyle}
        {...setTestId(getLocator(locator))}
      >
        {children}
      </TouchableOpacity>
    );
  }

  if (text) {
    return (
      <AnchorView
        accessibilityRole={accessibilityRole}
        accessible
        onPress={onPress || openUrl}
        accessibilityLabel={accessibilityLabel || text}
        style={customStyle}
        justifyContent={justifyContent}
        {...setTestId(getLocator(locator))}
        hitSlop={hitSlop}
      >
        <StyledText anchorVariation={anchorVariation} {...otherProps} {...styleProp}>
          {text}
        </StyledText>
        {visible && <AnchorIcon source={Icon} />}
      </AnchorView>
    );
  }

  return null;
};

Anchor.propTypes = {
  anchorVariation: PropTypes.string,
  text: PropTypes.string,
  visible: PropTypes.bool,
  children: PropTypes.shape({}),
  customStyle: PropTypes.shape({}),
  locator: PropTypes.string,
  onPress: PropTypes.func,
  accessibilityLabel: PropTypes.string,
  colorName: PropTypes.string,
  margins: PropTypes.string,
  justifyContent: PropTypes.string,
  hitSlop: PropTypes.shape({}),
  overrideStyle: PropTypes.shape({}),
};

Anchor.defaultProps = {
  anchorVariation: '',
  text: '',
  visible: false,
  children: null,
  customStyle: {},
  locator: '',
  onPress: null,
  accessibilityLabel: '',
  colorName: null,
  margins: null,
  justifyContent: '',
  hitSlop: {},
  overrideStyle: {},
};

export default withStyles(Anchor, AnchorStyles);
export { Anchor as AnchorVanilla };
