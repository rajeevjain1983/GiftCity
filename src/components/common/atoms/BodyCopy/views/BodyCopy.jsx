// 9fbef606107a605d69c0edbcd8029e5d
import React from "react";
import { PropTypes } from "prop-types";
import withStyles from "../../../hoc/withStyle";
import styles from "../BodyCopy.style";

const renderComponent = (compProps) => {
  const {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others
  } = compProps;
  return (
    <Component
      className={className}
      data-locator={dataLocator}
      style={customStyle}
      {...others}
    >
      {children}
    </Component>
  );
};

const BodyCopy = (props) => {
  const {
    children,
    className,
    fontFamily,
    component: Component,
    fontSize,
    fontWeight,
    lineHeight,
    letterSpacing,
    textAlign,
    color,
    dataLocator,
    link,
    customStyle,
    ...others
  } = props;

  const compProps = {
    Component,
    className,
    dataLocator,
    children,
    customStyle,
    ...others,
  };
  if (!link) {
    return renderComponent(compProps);
  }
};

const fontFamilyPropTypes = PropTypes.oneOf(["primary", "secondary"]);
const fontWeightPropTypes = PropTypes.oneOf([
  "regular",
  "semibold",
  "extrabold",
  "black",
]);
const letterSpacingPropTypes = PropTypes.oneOf([
  "ls271",
  "ls257",
  "ls222",
  "ls167",
  "normal",
]);
const textAlignPropTypes = PropTypes.oneOf(["left", "center", "right"]);
const colorPropTypes = PropTypes.oneOf([
  "text.primary",
  "text.secondary",
  "text.hint",
  "text.disabled",
  "white",
]);
const fontSizesPropTypes = PropTypes.oneOf([
  "fs48",
  "fs42",
  "fs36",
  "fs32",
  "fs28",
  "fs24",
  "fs22",
  "fs18",
  "fs16",
  "fs15",
  "fs14",
  "fs12",
  "fs10",
  "fs9",
]);

BodyCopy.defaultProps = {
  component: "p",
  className: null,
  children: null,
  fontFamily: "primary",
  fontSize: 12,
  fontWeight: "regular",
  lineHeight: "normal",
  letterSpacing: "normal",
  textAlign: "left",
  color: "text.primary",
  link: null,
  customStyle: {},
};

BodyCopy.propTypes = {
  component: PropTypes.elementType,
  className: PropTypes.string,
  children: PropTypes.node,
  fontFamily: PropTypes.oneOfType([
    PropTypes.arrayOf(fontFamilyPropTypes),
    fontFamilyPropTypes,
  ]),
  fontSize: PropTypes.oneOfType([
    PropTypes.arrayOf(fontSizesPropTypes),
    fontSizesPropTypes,
  ]),
  fontWeight: PropTypes.oneOfType([
    PropTypes.arrayOf(fontWeightPropTypes),
    fontWeightPropTypes,
  ]),
  lineHeight: PropTypes.oneOf(["normal"]),
  letterSpacing: PropTypes.oneOfType([
    PropTypes.arrayOf(letterSpacingPropTypes),
    letterSpacingPropTypes,
  ]),
  textAlign: PropTypes.oneOfType([
    PropTypes.arrayOf(textAlignPropTypes),
    textAlignPropTypes,
  ]),
  color: PropTypes.oneOfType([
    PropTypes.arrayOf(colorPropTypes),
    colorPropTypes,
  ]),
  dataLocator: PropTypes.string,
  link: PropTypes.shape({
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    target: PropTypes.string,
    text: PropTypes.string,
  }),
  customStyle: PropTypes.shape({}),
};

const StyledBodyCopy = withStyles(BodyCopy, styles);
StyledBodyCopy.defaultProps = BodyCopy.defaultProps;

export { BodyCopy as BodyCopyVanilla };
export default StyledBodyCopy;
