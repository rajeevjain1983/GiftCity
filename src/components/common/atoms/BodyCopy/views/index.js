// 9fbef606107a605d69c0edbcd8029e5d 
export { default } from './BodyCopy';
