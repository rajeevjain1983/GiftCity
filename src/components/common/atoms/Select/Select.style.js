// 9fbef606107a605d69c0edbcd8029e5d 
import { css } from 'styled-components';
import { getIconPath } from '@tcp/core/src/utils';

const downArrowIcon = getIconPath('down_arrow_icon');

const styleBottomSpacing = '26px';
const selectStyles = css`
  position: relative;
  display: block;
  height: 73px;

  .select__input {
    appearance: none;
    background: url(${downArrowIcon}) no-repeat right 0px bottom
      ${props => props.theme.spacing.ELEM_SPACING.SM};
    border: 0;
    border-bottom: 1px solid ${props => props.theme.colors.FOOTER.DIVIDER};
    ${props =>
      props.meta && props.meta.touched && props.meta.error
        ? `border-bottom: 1px solid ${props.theme.colors.TEXTBOX.ERROR_BORDER};`
        : ''};
    border-radius: 0;
    color: ${props =>
      props.disabled ? props.theme.colorPalette.gray[600] : props.theme.colors.TEXTBOX.COLOR};
    margin: 0;
    margin-bottom: ${props => props.theme.spacing.ELEM_SPACING.XS};
    outline: 0;
    padding: ${styleBottomSpacing} 0 6px;
    width: 100%;

    ${props =>
      props.disabled
        ? `
      background-color: ${props.theme.fieldBackgroundDisabledColor};
      border-color: ${props.theme.fieldBorderDisabledColor};
    `
        : ''};

    font-size: ${props => props.theme.typography.fontSizes.fs16};
  }

  .select__label {
    font-size: ${props => props.theme.fonts.fontSize.body.small.primary}px;
    font-weight: ${props => props.theme.fonts.fontWeight.bold};
    padding: 0;
    position: absolute;
    top: 0;
  }

  .SelectBox__error {
    display: flex;
    flex-direction: row;
    padding-bottom: ${props => props.theme.spacing.ELEM_SPACING.XXS};
  }

  .warning-icon {
    background: transparent url(${getIconPath('circle-alert-fill')}) no-repeat 0 0;
    background-size: contain;
    border: none;
    height: 14px;
    width: 16px;
    margin-right: 7px;
    flex-shrink: 0;
  }

  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default selectStyles;
