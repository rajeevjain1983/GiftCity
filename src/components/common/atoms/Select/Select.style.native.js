// 9fbef606107a605d69c0edbcd8029e5d 
import styled from 'styled-components';
import BodyCopy from '../BodyCopy';

export const StyledHeading = styled(BodyCopy)`
  margin-bottom: ${props => (props.isFocussed ? '0' : props.theme.spacing.ELEM_SPACING.XXS)};
  top: ${props => (props.isFocussed ? '0' : props.theme.spacing.ELEM_SPACING.MED)};
  position: absolute;
`;

export const Container = styled.View`
  position: relative;
  ${props => (props.isAddNewCC ? '' : `height: 72px`)};
`;

export default StyledHeading;
