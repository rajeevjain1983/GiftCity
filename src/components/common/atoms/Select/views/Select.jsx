// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import BodyCopy from '../../BodyCopy';
import withStyles from '../../../hoc/withStyles';
import StyledTextBox from '../Select.style';

const renderTitle = ({ value, title }) => {
  return value && title ? (
    <BodyCopy fontSize="fs12" fontFamily="secondary" className="select__label">
      {title}
    </BodyCopy>
  ) : (
    ''
  );
};

renderTitle.propTypes = {
  value: PropTypes.string,
  title: PropTypes.string,
};

renderTitle.defaultProps = {
  value: '',
  title: '',
};

// eslint-disable-next-line complexity
const SelectBox = ({
  className,
  id,
  ariaLabel,
  name,
  placeholder,
  input,
  options,
  meta: { touched, error },
  dataLocator,
  disabled,
  title,
  preText,
  forceSelect,
  ...otherProps
}) => {
  const elementID = id || name;
  const errorMessagea11yLbl = elementID ? `selectbox__error__${elementID}` : 'selectbox__error';
  return (
    <div className={className}>
      <select
        {...input}
        id={id}
        aria-label={ariaLabel || title || placeholder}
        className="select__input"
        name={name}
        value={input.value || placeholder}
        data-locator={dataLocator}
        {...otherProps}
        disabled={disabled}
        aria-describedby={touched && errorMessagea11yLbl}
      >
        {forceSelect && <option value="">{placeholder}</option>}
        {!input.value && placeholder && <option value="">{placeholder}</option>}
        {options &&
          options.map(option => {
            return (
              <option
                value={option.id || option.get('id')}
                id={`${elementID}_${option.id || option.get('id')}`} // When month and day dropdown is on same page id from 10 - 12 was getting duplicate
                key={option.id || option.get('id')}
              >
                {preText}
                {option.displayName || option.get('displayName')}
              </option>
            );
          })}
      </select>
      {/* the placeholder should not be used as title since different text
      is being used for placeholder and title in multiple instances */}
      {renderTitle({ value: input.value, title })}
      <div className="SelectBox__error">
        <div className={touched && error ? 'warning-icon' : ''} aria-disabled="true" />
        {touched && error && (
          <BodyCopy
            fontSize="fs12"
            fontFamily="secondary"
            component="div"
            color="error"
            fontWeight="extrabold"
            role="alert"
            aria-live="assertive"
            id={errorMessagea11yLbl}
          >
            {error}
          </BodyCopy>
        )}
      </div>
    </div>
  );
};

SelectBox.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  ariaLabel: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  placeholder: PropTypes.string,
  meta: PropTypes.shape({}).isRequired,
  input: PropTypes.shape({}).isRequired,
  options: PropTypes.shape([]).isRequired,
  dataLocator: PropTypes.string,
  disabled: PropTypes.bool,
  title: PropTypes.string,
  preText: PropTypes.string,
  forceSelect: PropTypes.bool,
};

SelectBox.defaultProps = {
  id: '',
  ariaLabel: '',
  name: '',
  type: 'text',
  placeholder: '',
  dataLocator: '',
  disabled: false,
  title: '',
  className: '',
  preText: '',
  forceSelect: '',
};

export default withStyles(SelectBox, StyledTextBox);
export { SelectBox as SelectBoxVanilla };
