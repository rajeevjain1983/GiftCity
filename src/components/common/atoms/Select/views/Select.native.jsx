// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import PropTypes from 'prop-types';
import { StyledErrorWrapper, ViewWithSpacing } from '../../styledWrapper/styledWrapper.native';
import { StyledHeading, Container } from '../Select.style.native';
import NativeDropdown from '../../NativeDropDown';
import BodyCopy from '../../BodyCopy';
import Image from '../../Image';

const errorIcon = require('../../../../../../../mobileapp/src/assets/images/alert-triangle.png');

const displayHeading = (input, placeholder, heading) => {
  const isFocussed = !!input.value || !!placeholder;
  return (
    !!input.value && (
      <StyledHeading
        isFocussed={isFocussed}
        fontSize={isFocussed ? 'fs10' : 'fs14'}
        fontWeight={isFocussed ? 'extrabold' : 'regular'}
        text={heading}
      />
    )
  );
};

const SelectBox = ({
  id,
  name,
  input,
  options,
  heading,
  onValueChange,
  placeholder,
  meta: { touched, error },
  isAddNewCC,
  pickerStyle,
  ...otherProps
}) => {
  if (placeholder && !input.value && options && options[0].displayName !== placeholder) {
    options.unshift({
      displayName: placeholder,
      id: '',
    });
  }
  return (
    <Container isAddNewCC={isAddNewCC}>
      {displayHeading(input, placeholder, heading)}
      <NativeDropdown
        onValueChange={value => {
          input.onChange(value);
          if (input.onBlur) input.onBlur(value);
          onValueChange(value);
        }}
        data={options}
        selectedValue={input.value}
        id={id}
        name={name}
        buttonVariation="mobileApp-select"
        width="100%"
        pickerStyleObj={pickerStyle}
        {...otherProps}
      />
      {touched && error && (
        <StyledErrorWrapper>
          <ViewWithSpacing spacingStyles="margin-right-XS">
            <Image source={errorIcon} alt="" width="16px" height="14px" />
          </ViewWithSpacing>
          <BodyCopy
            fontWeight="extrabold"
            fontFamily="secondary"
            fontSize="fs12"
            text={error}
            color="error"
          />
        </StyledErrorWrapper>
      )}
    </Container>
  );
};

SelectBox.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  input: PropTypes.shape({}).isRequired,
  options: PropTypes.shape([]).isRequired,
  onValueChange: PropTypes.func,
  heading: PropTypes.string,
  placeholder: PropTypes.string,
  meta: PropTypes.shape({
    touched: PropTypes.bool,
    error: PropTypes.bool,
  }),
  isAddNewCC: PropTypes.bool,
  pickerStyle: PropTypes.shape({}),
};

SelectBox.defaultProps = {
  id: '',
  placeholder: '',
  onValueChange: () => {},
  heading: '',
  meta: {},
  isAddNewCC: false,
  pickerStyle: {},
};

export default SelectBox;
