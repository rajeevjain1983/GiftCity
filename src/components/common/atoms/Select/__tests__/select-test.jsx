// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { SelectBoxVanilla } from '../views/Select';

describe('Selectbox component', () => {
  const countryUS = 'United States';
  const displayValue = [
    {
      id: 'US',
      displayName: countryUS,
    },
    { id: 'CA', displayName: 'Canada' },
  ];
  it('renders correctly', () => {
    const props = {
      name: '',
      placeholder: 'placeholder',
      Value: 'hi',
      type: 'text',
      id: 'abcd',
      className: 'asdfasdf',
      options: displayValue,
      meta: {
        touched: '',
        error: '',
        warning: '',
      },
      input: {
        value: 'hello',
      },
    };
    const component = shallow(<SelectBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with disabled props', () => {
    const props = {
      name: '',
      placeholder: 'placeholder',
      Value: 'hi',
      type: 'text',
      id: 'abcd',
      className: 'asdfasdf',
      options: displayValue,
      meta: {
        touched: '',
        error: '',
        warning: '',
      },
      input: {
        value: 'hello',
      },
      disabled: true,
    };
    const component = shallow(<SelectBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with touched and error', () => {
    const valueInfo = 'some string @@LABEL@@';
    const props = {
      id: 'abcd',
      className: 'asdfasdf',
      options: displayValue,
      value: valueInfo,
      meta: {
        touched: true,
        error: 'REQUIRED_FIELD',
        warning: '',
      },
      input: {},
      defaultValue: countryUS,
      placeholder: 'first name for user',
    };
    const component = shallow(<SelectBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with touched and warning', () => {
    const props = {
      type: 'text',
      id: 'abcd',
      className: 'asdfasdf',
      options: displayValue,
      meta: {
        touched: true,
        error: null,
        warning: 'this is a warning',
      },
      input: {},
      defaultValue: countryUS,
    };
    const component = shallow(<SelectBoxVanilla {...props} />);
    expect(component).toMatchSnapshot();
  });
});
