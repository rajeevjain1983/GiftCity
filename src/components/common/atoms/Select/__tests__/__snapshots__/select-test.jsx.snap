// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectbox component renders correctly 1`] = `
<div
  className="asdfasdf"
>
  <select
    Value="hi"
    aria-describedby=""
    aria-label="placeholder"
    className="select__input"
    data-locator=""
    disabled={false}
    id="abcd"
    name=""
    type="text"
    value="hello"
  >
    <option
      id="abcd_US"
      key="US"
      value="US"
    >
      United States
    </option>
    <option
      id="abcd_CA"
      key="CA"
      value="CA"
    >
      Canada
    </option>
  </select>
  <div
    className="SelectBox__error"
  >
    <div
      aria-disabled="true"
      className=""
    />
  </div>
</div>
`;

exports[`Selectbox component renders correctly with disabled props 1`] = `
<div
  className="asdfasdf"
>
  <select
    Value="hi"
    aria-describedby=""
    aria-label="placeholder"
    className="select__input"
    data-locator=""
    disabled={true}
    id="abcd"
    name=""
    type="text"
    value="hello"
  >
    <option
      id="abcd_US"
      key="US"
      value="US"
    >
      United States
    </option>
    <option
      id="abcd_CA"
      key="CA"
      value="CA"
    >
      Canada
    </option>
  </select>
  <div
    className="SelectBox__error"
  >
    <div
      aria-disabled="true"
      className=""
    />
  </div>
</div>
`;

exports[`Selectbox component renders correctly with touched and error 1`] = `
<div
  className="asdfasdf"
>
  <select
    aria-describedby="selectbox__error__abcd"
    aria-label="first name for user"
    className="select__input"
    data-locator=""
    defaultValue="United States"
    disabled={false}
    id="abcd"
    name=""
    type="text"
    value="some string @@LABEL@@"
  >
    <option
      value=""
    >
      first name for user
    </option>
    <option
      id="abcd_US"
      key="US"
      value="US"
    >
      United States
    </option>
    <option
      id="abcd_CA"
      key="CA"
      value="CA"
    >
      Canada
    </option>
  </select>
  <div
    className="SelectBox__error"
  >
    <div
      aria-disabled="true"
      className="warning-icon"
    />
    <Styled(BodyCopy)
      aria-live="assertive"
      className={null}
      color="error"
      component="div"
      customStyle={Object {}}
      fontFamily="secondary"
      fontSize="fs12"
      fontWeight="extrabold"
      id="selectbox__error__abcd"
      letterSpacing="normal"
      lineHeight="normal"
      link={null}
      role="alert"
      textAlign="left"
    >
      REQUIRED_FIELD
    </Styled(BodyCopy)>
  </div>
</div>
`;

exports[`Selectbox component renders correctly with touched and warning 1`] = `
<div
  className="asdfasdf"
>
  <select
    aria-describedby="selectbox__error__abcd"
    aria-label=""
    className="select__input"
    data-locator=""
    defaultValue="United States"
    disabled={false}
    id="abcd"
    name=""
    type="text"
    value=""
  >
    <option
      id="abcd_US"
      key="US"
      value="US"
    >
      United States
    </option>
    <option
      id="abcd_CA"
      key="CA"
      value="CA"
    >
      Canada
    </option>
  </select>
  <div
    className="SelectBox__error"
  >
    <div
      aria-disabled="true"
      className=""
    />
  </div>
</div>
`;
